source/Ship.cpp

```diff
@@ -3174,7 +3159,7 @@ bool Ship::CanCarry(const Ship &ship) const
 
 void Ship::AllowCarried(bool allowCarried)
 {
-	canBeCarried = allowCarried && BAY_TYPES.count(attributes.Category()) > 0;
+	canBeCarried = allowCarried && GameData::CategorySet("bay type").count(attributes.Category()) > 0;
```

Comment:
Converting to set looks a little bit inefficient for a single lookup, I suggest to use find directly on the vector (also on other places in this PR).
```suggestion
	bayCategory = GameData::CategorySet("bay type");
	canBeCarried = allowCarried && (find(bayCategory.begin(), bayCategory.end(), attributes.Category()) != bayCategory.end());
```

Reply:
find_if?